### PR TITLE
Require user to be shown the developer agreement before allowing them to sign it (bug 1210191).

### PR DESCRIPTION
**Pull request by chuckharmston**
r? @ngokevin

### CODE REVIEW

File: mkt/users/models.py [hunk 1]
```diff
@@ -83,6 +82,9 @@ class UserProfile(OnChangeMixin, ModelBase, AbstractBaseUser):
                             editable=False)
     enable_recommendations = models.BooleanField(default=True)
 
+    shown_dev_agreement = models.DateTimeField(null=True, blank=True)
+    read_dev_agreement = models.DateTimeField(null=True, blank=True)
```
Comment (ngokevin) [inline mkt/users/models.py]: shown vs read? maybe shown vs signed?

File: docs/api/topics/accounts.rst [hunk 2]
```diff
@@ -237,9 +237,29 @@ carrier/region pairs.
...
+    :status 200: successfully viewed developer agreement.
+    :status 201: successfully viewed developer agreement for the first time.
+      The user can now :ref:`sign the agreement <read-agreement>`.
+    :status 400: user has already signed terms of service.
+    :status 403: authentication required.
+    :status 405: invalid HTTP method; only POST is allowed on this endpoint.
+
+.. read-agreement:
+
+.. http:post:: /api/v2/account/dev-agreement/read/
```
Comment (ngokevin) [inline docs/api/topics/accounts.rst]: feel like s/read/sign makes sense

File: mkt/users/models.py (hunk 1 above)
Comment (chuckharmston) [inline mkt/users/models.py]: That's what I'd prefer, but I didn't want to change wording with the existing field, which is `read` :/

Comment (ngokevin) [inline mkt/users/models.py]: hmm...add comments at least

**Comment by ngokevin** [comment]
r+wc

**Pull request merged**